livesql/binlog.go

```diff
@@ -188,24 +188,30 @@ func buildColumnMap(conn *sql.DB, database string, table *sqlgen.Table) (*column
 
 // parseBinlogRow parses a binlog row into a struct
 func parseBinlogRow(table *sqlgen.Table, binlogRow []interface{}, columnMap *columnMap) (interface{}, error) {
+	ptr := reflect.New(table.Type)
+	elem := ptr.Elem()
+
 	if len(binlogRow) != columnMap.expectedColumns {
 		return nil, fmt.Errorf("binlog for %s has %d columns, expected %d",
 			table.Name, len(binlogRow), columnMap.expectedColumns)
 	}
 
-	scanners := make([]*fields.Scanner, len(columnMap.source))
 	for i, j := range columnMap.source {
 		if j == -1 {
 			continue
 		}
-		scanner := table.Columns[i].Descriptor.Scanner()
+		column := table.Columns[i]
+		field := elem.FieldByIndex(column.Index)
+		if field.Kind() != reflect.Ptr && field.CanAddr() {
```

Comment:
this this alright when it's not a pointer and we can't Addr it?  Or is that not physically possible?

Reply:
It _should_ not be possible. However, you're right - if it's not a pointer and we can't addr it, we should return an error here.

Reply:
Hmm, actually, maybe I should remove `field.CanAddr()` and just have it panic if it can't. This should _always_ be able to Addr - the struct is even defined in this method